File: langgraph/data/semantic_equiv/8.py
from typing import List, Tuple


def calculate_aggregates(values: List[int]) -> Tuple[int, int]:
    total_sum = 0
    cumulative_product = 1

    index = 0
    while index < len(values):
        total_sum += values[index]
        cumulative_product *= values[index]
        index += 1

    return total_sum, cumulative_product

# method name: sum_product -> calculate_aggregates
# parameter name: numbers -> values
# variable name: sum_value -> total_sum, prod_value -> cumulative_product
# replaced for loop with while loop

File: langgraph/result/qwen/3b-eq2js/8.js
function calculateAggregates(values) {
    let totalSum = 0;
    let cumulativeProduct = 1;

    let index = 0;
    while (index < values.length) {
        totalSum += values[index];
        cumulativeProduct *= values[index];
        index++;
    }

    return [totalSum, cumulativeProduct];
}

// method name: sum_product -> calculateAggregates
// parameter name: numbers -> values
// variable name: sum_value -> totalSum, prod_value -> cumulativeProduct
// replaced for loop with while loop